### ISSUE TITLE
Product Edit Page Can't Load

### ISSUE THREAD

**Issue by Nerogee** [magento/magento2#5967]
### Preconditions

1. Magento 2.1
2. php 7.0.7
3. MariaDB 10.1
### Steps to reproduce

1. Delete:   rm -R  pub/static/*
2. Delete:   rm -R var/view_preprocessed/*
3. Deploy static file: bin/magento setup:static-content:deploy
4. Go do any backend product edit page
### Expected result

1. Product Edit page load successfully
### Actual result

1. [Screenshot, logs]
2. There are many javascript error
   
   ![bugshoot](https://cloud.githubusercontent.com/assets/12842026/17372874/41bef7de-59d8-11e6-88f9-5cd62bad8216.png)

**Comment by orlangur** [comment]
Try to click on those problematic lines to open *.js files - as there are some syntax errors, looks like they are not fully loaded from web server.

**Comment by Nerogee** [comment]
Here is a broken function from i18n.js

execute = function (element, valueAccessor, isUpdate) {
            var original = ko.unwrap(valueAccessor() || ''),
                el = getRealElement(element, isUpdate),
                inlineTranslation = (module.config() || {}).inlineTranslation;

```
        if (inlineTranslation) {
            setTranslateProp(el, original);
        } else {
            setText(el, "'"
    .addAttribute('translate', {
        binding: 'i18n'
    });
```

yes, it is. The newly generated file seem to be broken. @orlangur do you have solution. I have newly installed magento2.1 which seems to be working well. So I don't know what caused this issue? My products data ? or some implicit configuration? To mention here, I don't enable any JS binding/merging/ minifying at all.

**Comment by orlangur** [comment]
Do materialized files in `pub/static` correspond to original file versions in your installation?

It's hard to say why your browser receives incorrect version, it could be web server misconfiguration or some malfunction in browser cache. For example, such problem can be easily emulated with no free space on device.

**Comment by Nerogee** [comment]
At first I did use signed static file, but I can't find materialized files. I realized that function may cause this issue so that I turned it off. But no matter how, It still has error. BTW, how do you know my browser receives incorrect version? As I already noticed, the newly generated file is not the same one of the origin. What should I check further? thanks

**Comment by Nerogee** [comment]
Hi @orlangur. Could you please give me hints on why this issue happen and what should I do to solve. 
As you mentioned above it may because of no free space, but I don't think it's the case. My reason is simply because there are lots of space and this file looks complete. Saying so because the final ending **})** is there. The file is broken in the middle way.
 So I highly suspect it's the function that generated it got glitch. 

 Any suggestion would be appreciated.

**Comment by Nerogee** [comment]
Finally, I found out the reason.. The **"translate_strategy"** is culprit. I set it to **embedded** which caused this issue. Cost me almost a daytime to find it out. hope can help others who suffer the same.

**Comment by lucasalfaro** [comment]
uff.. Thanks you are my hero of the day. After an upgrade for mage 2.1 i had several error of JS, that one being one I had no idea how to tackle. Thanks again.

**Comment by slopukhov** [comment]
@Nerogee thank you for your feedback.
Please, recheck it on the latest 2.1 version of Magento.

**Comment by vzabaznov** [comment]
Hi @Nerogee , closed as we don't receive update more than 2 week, feel free to write here/reopen or to open new issue

**Comment by devdesco-ceo** [comment]
Hi @Nerogee,
I can confirm that i18n.js gets broken by being concatenated/minified even if JS Minification is turned off. I went through the minified code and the lines in the end of both `adminhtml` and `frontend` side i18n.js must be replaced as follows (either manually, or by another method like Nginx substitute for a temporary fix):

BEFORE:

`            if (inlineTranslation) {
                 setTranslateProp(el, original);
             } else {
                 setText(el, "'"
         .addAttribute('translate', {
             binding: 'i18n'
         });
});`


AFTER:

`         if (inlineTranslation) {
                 setTranslateProp(el, original);
             } else {
                 setText(el, "'"
         .addAttribute('translate', {
             binding: 'i18n'
         }));
     }};
 });`

**Comment by orlangur** [comment]
@viktor-zhuromskyy, nice finding, but closed issues are most likely not monitored by Magento folks. Please report a new one if it is still relevant.

**Comment by maghamed** [comment]
issue is reopened based on @viktor-zhuromskyy  reply

**Comment by IlnitskiyArtem** [comment]
Hi, @Nerogee.
We've created internal ticket MAGETWO-70599 to address this issue.

**Comment by magento-engcom-team** [comment]
@Nerogee, thank you for your report.
We've created internal ticket(s) MAGETWO-70599 to track progress on the issue.

**Comment by magento-engcom-team** [comment]
Hi @Nerogee. 

Thank you for your report and collaboration!

The issue was fixed by Magento team. The fix was delivered into `magento/magento2:2.3-develop` branch(es).
Related commit(s):
 - [71ce98baa01c63e59f1664a83c6ff6e41bc58c29](https://github.com/magento/magento2/commit/71ce98baa01c63e59f1664a83c6ff6e41bc58c29)
 - [c56d933c35e49b6f00dda23ef8a34c148c803c9e](https://github.com/magento/magento2/commit/c56d933c35e49b6f00dda23ef8a34c148c803c9e)
 - [26409acc22e86548d53a8127ff7c4417ab998336](https://github.com/magento/magento2/commit/26409acc22e86548d53a8127ff7c4417ab998336)
 - [29361a3b615c64008c29620c5dff6c71c792a3c1](https://github.com/magento/magento2/commit/29361a3b615c64008c29620c5dff6c71c792a3c1)
 - [a0e2f0d4d0c4d493f47f51d31989cbc888dd5fe5](https://github.com/magento/magento2/commit/a0e2f0d4d0c4d493f47f51d31989cbc888dd5fe5)
 - [d68fe438f39d99738e9adde19410c75ff5d4682a](https://github.com/magento/magento2/commit/d68fe438f39d99738e9adde19410c75ff5d4682a)
 - [3c524e3aefb06344f2703e80fe6fc2bc0b40b36e](https://github.com/magento/magento2/commit/3c524e3aefb06344f2703e80fe6fc2bc0b40b36e)
 - [61741bc94ac5f197043275ce505117134ab70970](https://github.com/magento/magento2/commit/61741bc94ac5f197043275ce505117134ab70970)
 - [f00cbe83bc28574edfa1ccfcb060a13f5aab78fa](https://github.com/magento/magento2/commit/f00cbe83bc28574edfa1ccfcb060a13f5aab78fa)

The fix will be available with the upcoming `2.3.3` release.
